frontend/lib/swr-cache/index.ts

```diff
@@ -72,18 +72,29 @@ export const withCache = <
    const logger: Logger = defaultLog;
 
    const requestRevalidation = async (variables: z.infer<VariablesSchema>) => {
-      fetch(`${APP_ORIGIN}/${endpoint}`, {
-         method: 'POST',
-         body: JSON.stringify(variables),
-      }).catch((error) => {
-         logger.error(
-            `${endpoint}.error: failed to trigger revalidation\n${printError(
-               error
-            )}`
-         );
+      await new Promise((resolve, reject) => {
+         // Don't delay the lambda beyond 50ms for revalidation
+         setTimeout(resolve, 50);
```

Comment:
Do we need to cancel this if the request finishes first? i.e. Does calling `resolve()` twice cause any problems?

Reply:
@sterlinghirsh had the same question! I don't think so, this always produces `'done'` when I run it with node

```js
(async () => {
   const p = new Promise((resolve, reject) => {
      resolve('done');
      resolve('second resolution');
      throw new Error('oh no!');
      reject('rejection');
   });
   const res = await p;
   console.log(res);
})();
```

MDN also [suggests](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/Promise#effect_of_calling_resolvefunc) that calling it twice has no effect.

> Calling resolveFunc causes the promise to become resolved, so that calling resolveFunc or rejectFunc again has no effect.

Reply:
That being said, I'm happy to clean up the timeout if we don't like this pattern.

Reply:
It's fine by me to leave it.